nixos/modules/services/network-filesystems/moosefs.nix

```diff
@@ -205,42 +213,86 @@ in {
       };
     };
 
-    # Create system user account for daemons
-    users = lib.mkIf ( cfg.runAsUser && ( cfg.master.enable || cfg.metalogger.enable || cfg.chunkserver.enable ) ) {
+    users = lib.mkIf ( cfg.runAsUser && ( cfg.master.enable || cfg.metalogger.enable || cfg.chunkserver.enable || cfg.cgiserver.enable ) ) {
       users.moosefs = {
         isSystemUser = true;
-        description = "moosefs daemon user";
+        description = "MooseFS daemon user";
         group = "moosefs";
+        home = "/var/lib/mfs";
+        createHome = true;
       };
       groups.moosefs = {};
     };
 
-    environment.systemPackages =
-      (lib.optional cfg.client.enable pkgs.moosefs) ++
-      (lib.optional cfg.master.enable initTool);
-
-    networking.firewall.allowedTCPPorts =
-      (lib.optionals cfg.master.openFirewall [ 9419 9420 9421 ]) ++
-      (lib.optional cfg.chunkserver.openFirewall 9422);
-
-    # Ensure storage directories exist
-    systemd.tmpfiles.rules =
-         lib.optional cfg.master.enable "d ${cfg.master.settings.DATA_PATH} 0700 ${mfsUser} ${mfsUser}"
-      ++ lib.optional cfg.metalogger.enable "d ${cfg.metalogger.settings.DATA_PATH} 0700 ${mfsUser} ${mfsUser}"
-      ++ lib.optional cfg.chunkserver.enable "d ${cfg.chunkserver.settings.DATA_PATH} 0700 ${mfsUser} ${mfsUser}";
-
-    # Service definitions
-    systemd.services.mfs-master = lib.mkIf cfg.master.enable
-    ( systemdService "master" {
-      TimeoutStartSec = 1800;
-      TimeoutStopSec = 1800;
-      Restart = "no";
-    } masterCfg );
-
-    systemd.services.mfs-metalogger = lib.mkIf cfg.metalogger.enable
-      ( systemdService "metalogger" { Restart = "on-abnormal"; } metaloggerCfg );
-
-    systemd.services.mfs-chunkserver = lib.mkIf cfg.chunkserver.enable
-      ( systemdService "chunkserver" { Restart = "on-abnormal"; } chunkserverCfg );
-    };
+    environment.systemPackages = lib.optional cfg.client.enable pkgs.moosefs;
+
+    networking.firewall.allowedTCPPorts = lib.mkMerge [
+      (lib.optionals cfg.master.openFirewall [ 9419 9420 9421 ])
+      (lib.optional cfg.chunkserver.openFirewall 9422)
+      (lib.optional (cfg.cgiserver.enable && cfg.cgiserver.openFirewall) cfg.cgiserver.settings.PORT)
+    ];
+
+    systemd.tmpfiles.rules = [
+      # Base directory for MooseFS
+      "d /var/lib/mfs 0755 ${mfsUser} ${mfsUser} -"
+      "d /run/mfs 0755 ${mfsUser} ${mfsUser} -"
```

Comment:
Why do we need to create these two directories?

Reply:
Not need now because have make a workaround.